### PR TITLE
Notify command explicitly

### LINKED ISSUE #13684: Debugger invokes `notifyCommandChanged` force-updating commands after tab switch
## Description

The debugger extension plugins for notebook, console and file all invoke a full commands update, e.g.:

https://github.com/jupyterlab/jupyterlab/blob/edc294b06fe6607be80094dd590cb9f766ab83ff/packages/debugger-extension/src/index.ts#L138

resulting in a long task preventing fast consecutive tab switch.

## Reproduce

1. Open two notebooks
2. Record profile
3. Switch tabs
4. See a long task invoking React re-render as below:

Screenshot from 2022-12-29 17-02-45

## Expected behavior

There is no needless updates. Maybe something like;

https://github.com/jupyterlab/jupyterlab/blob/edc294b06fe6607be80094dd590cb9f766ab83ff/packages/toc-extension/src/index.ts#L303-L304

## Context

- JupyterLab version:  4.0.0a31

### PR DESCRIPTION
**Pull request by fcollonval**
## References

Fixes #13684

## Code changes

Call explicitly the commands of the extension and not all commands to avoid useless rerendering.

## User-facing changes

None

## Backwards-incompatible changes

None

### CODE REVIEW

**Review by krassowski — APPROVED** [review]
Thank you!

Ideally we would fix the debugger tests (#13903) before merging this one.

**Comment by jtpio** [comment]
> Ideally we would fix the debugger tests (#13903) before merging this one.

Just rebased with the GitHub UI: 

image

To grab the `js-debugger` fix from https://github.com/jupyterlab/jupyterlab/pull/13941

**Comment by jtpio** [comment]
The `js-debugger` check is now passing: 

image

**Review by jtpio — APPROVED** [review]
Thanks!

File: packages/debugger-extension/src/index.ts [hunk 1]
```diff
@@ -48,6 +48,12 @@ import { Session } from '@jupyterlab/services';
 import { ISettingRegistry } from '@jupyterlab/settingregistry';
 import { ITranslator } from '@jupyterlab/translation';
 
+function notifyCommands(app: JupyterFrontEnd): void {
```
Comment (jtpio) [inline packages/debugger-extension/src/index.ts:51]: Ah this seems to be triggering commands regardless whether the plugin is available or activated on the page.

Spotted downstream in Notebook 7, which does not include the command to inspect variables (`@jupyterlab/debugger-extension:variables`):

image

Comment (jtpio) [reply packages/debugger-extension/src/index.ts:51]: Opened https://github.com/jupyterlab/jupyterlab/issues/14023

**Pull request merged**